### PR TITLE
Move is_up into retry cycle

### PR DESCRIPTION
**Pull request by dkropachev**
https://trello.com/c/yKIK9yUc/1413-enhance-remotecmdrunner-to-make-ssh-connection-persisstent
## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I gave variables/functions meaningful self-explanatory names
- [x] I didn't leave commented-out/debugging code
- [x] I didn't copy-paste code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

**Comment by bentsi** [comment]
@dkropachev 
1) You haven't done the PR pre-checks properly
2) please run `test-provision-gce` (or `test-provision-aws`) to verify it works

**Comment by dkropachev** [comment]
> @dkropachev
> 
> 1. You haven't done the PR pre-checks properly
> 2. please run `test-provision-gce` (or `test-provision-aws`) to verify it works

1. Fixed
2. provision test passed, but pipelie failed due to the issue in Jenkins configuration, which is not related to this PR

File: sdcm/cluster.py [hunk 1]
```diff
@@ -2111,7 +2111,7 @@ def install_mgmt(self, scylla_repo, scylla_mgmt_repo, auth_token, segments_per_r
             except Exception as ex:  # pylint: disable=broad-except
                 self.log.warning(ex)
         else:
-            self.remoter.run('sudo scyllamgr_setup -y')
+            self.remoter.run('echo yes| sudo scyllamgr_setup')
```
Comment (bentsi) [inline sdcm/cluster.py]: @yarongilor did `scyllamgr_setup` setup changed?

Comment (yarongilor) [reply sdcm/cluster.py]: @bentsi , i think it will be better to confirm this with @ShlomiBalalis since he's writing tests for the new backup feature.

Comment (fgelcer) [reply sdcm/cluster.py]: @dkropachev , you better use as it was, ```sudo scyllamgr_setup -y```

Comment (bentsi) [reply sdcm/cluster.py]: @fgelcer why did you change it to `scyllamgr_setup -y`?

Comment (bentsi) [reply sdcm/cluster.py]: your change broke provision test

Comment (fgelcer) [reply sdcm/cluster.py]: because manager supports it, instead of the "trick" to ```echo yes```

Comment (fgelcer) [reply sdcm/cluster.py]: which manager version does it use?
we should move every usage of manager to 2.0

Comment (bentsi) [reply sdcm/cluster.py]: scylla_mgmt_repo: 'http://downloads.scylladb.com.s3.amazonaws.com/manager/rpm/unstable/centos/master/latest/scylla-manager.repo'

Comment (bentsi) [reply sdcm/cluster.py]: with Scylla 3.2.rc1

Comment (bentsi) [reply sdcm/cluster.py]: it used defaults: 
```
scylla_repo_m: 'http://repositories.scylladb.com/scylla/repo/qa-test/centos/scylladb-2019.1.repo'
scylla_mgmt_repo: 'http://downloads.scylladb.com.s3.amazonaws.com/manager/rpm/unstable/centos/master/latest/scylla-manager.repo'
```

Comment (bentsi) [reply sdcm/cluster.py]: from the `defaults/test_default.yaml`

Comment (bentsi) [reply sdcm/cluster.py]: 15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR > Cluster longevity-200gb-48h-network-master-monitor-set-4ff29c0e (AMI: ['ami-02eac2c0129f6376b'] Type: t3.large): Node setup failed: Node longevity-200gb-48h-network-master-monitor-node-4ff29c0e-1 [35.173.213.43 | 172.30.0.193] (seed: False) < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR > Cluster longevity-200gb-48h-network-master-monitor-set-4ff29c0e (AMI: ['ami-02eac2c0129f6376b'] Type: t3.large): Node setup failed: Node longevity-200gb-48h-network-master-monitor-node-4ff29c0e-1 [35.173.213.43 | 172.30.0.193] (seed: False)
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR > Traceback (most recent call last):
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR >   File "/sct/sdcm/cluster.py", line 2754, in node_setup
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR >     cl_inst.node_setup(node, **setup_kwargs)
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR >   File "/sct/sdcm/cluster.py", line 3806, in node_setup
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR >     self.install_scylla_manager(node, auth_token=self.mgmt_auth_token)
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR >   File "/sct/sdcm/cluster.py", line 3814, in install_scylla_manager
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR >     segments_per_repair=self.params.get('mgmt_segments_per_repair'))
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR >   File "/sct/sdcm/cluster.py", line 2114, in install_mgmt
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR >     self.remoter.run('sudo scyllamgr_setup -y')
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR >   File "/sct/sdcm/remote.py", line 261, in run
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR >     result = _run()
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR >   File "/sct/sdcm/utils/common.py", line 92, in inner
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR >     return func(*args, **kwargs)
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR >   File "/sct/sdcm/remote.py", line 248, in _run
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR >     watchers=watchers, timeout=timeout)
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR >   File "<decorator-gen-3>", line 2, in run
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR >   File "/usr/local/lib/python3.6/site-packages/fabric/connection.py", line 30, in opens
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR >     return method(self, *args, **kwargs)
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR >   File "/usr/local/lib/python3.6/site-packages/fabric/connection.py", line 721, in run
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR >     return self._run(self._remote_runner(), command, **kwargs)
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR >   File "/usr/local/lib/python3.6/site-packages/invoke/context.py", line 101, in _run
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR >     return runner.run(command, **kwargs)
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR >   File "/usr/local/lib/python3.6/site-packages/invoke/runners.py", line 291, in run
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR >     return self._run_body(command, **kwargs)
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR >   File "/usr/local/lib/python3.6/site-packages/invoke/runners.py", line 442, in _run_body
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR >     raise UnexpectedExit(result)
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR > invoke.exceptions.UnexpectedExit: Encountered a bad command exit code!
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR > 
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR > Command: 'sudo scyllamgr_setup -y'
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR > 
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR > Exit code: 1
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR > 
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR > Stdout:
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR > 
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR > unknown command -y
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR > 
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR > Stderr:
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR > 
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR > 
15:55:59  < t:2020-01-22 12:55:56,567 f:cluster.py      l:2756 c:sdcm.cluster         p:ERROR > 
15:55:59  < t:2020-01-22 12:55:56,587 f:tester.py       l:129  c:sdcm.tester          p:ERROR > Exception in setUp. Will call tearDown < t:2020-01-22 12:55:56,587 f:tester.py       l:129  c:sdcm.tester          p:ERROR > Exception

**Comment by dkropachev** [comment]
Provision test completed

**Pull request merged**